### PR TITLE
Add support for the `.gemini` connector type for Kibana action connectors

### LINKED ISSUE #727: [Bug] Google `.gemini` connector not recognized as valid connector id in elasticstack_kibana_action_connector
**Describe the bug**
Terraform can't spin up Gemini Connector for Elastic AI Assistant using `elasticstack_kibana_action_connector` resource.

**To Reproduce**
Assume that elastic connection is already set up and working

1. dummy terraform config
```terraform
resource "google_service_account" "gemini_sa" {
  account_id = "ec-observability-ai-assist"
  project = "<redacted>"
}

resource "google_service_account_key" "observability_gemini" {
  service_account_id = google_service_account.gemini_sa.id
}

resource "google_project_iam_member" "observability_gemini" {
  member  = google_service_account.gemini_sa.member
  project = "<redacted>"
  role    = "roles/aiplatform.user"
}

resource "elasticstack_kibana_action_connector" "observability_gemini_connector" {
  provider = elasticstack.observability
  connector_type_id = ".gemini"
  name              = "gemini-playground"
  config = jsonencode({
    apiUrl = "https://europe-west3-aiplatform.googleapis.com"
    defaultModel = "gemini-1.5-pro-001"
    gcpProjectID = "<redacted>"
    gcpRegion = "europe-west3"
  })
  secrets = base64decode(google_service_account_key.gemini_sa.private_key)
}

```
2. `terraform apply` fails with
```
  # elasticstack_kibana_action_connector.observability_gemini_connector will be created
  + resource "elasticstack_kibana_action_connector" "observability_gemini_connector" {
      + config             = jsonencode(
            {
              + apiUrl       = "https://europe-west3-aiplatform.googleapis.com"
              + defaultModel = "gemini-1.5-pro-001"
              + gcpProjectID = "<redacted>"
              + gcpRegion    = "europe-west3"
            }
        )
      + connector_id       = (known after apply)
      + connector_type_id  = ".gemini"
      + id                 = (known after apply)
      + is_deprecated      = (known after apply)
      + is_missing_secrets = (known after apply)
      + is_preconfigured   = (known after apply)
      + name               = "gemini-playground"
      + secrets            = (sensitive value)
      + space_id           = "default"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

elasticstack_kibana_action_connector.observability_gemini_connector: Creating...
╷
│ Error: unknown connector type [.gemini]
│ 
│   with elasticstack_kibana_action_connector.observability_gemini_connector,
│   on observability_elastic_ai_assistant.tf line 18, in resource "elasticstack_kibana_action_connector" "observability_gemini_connector":
│   18: resource "elasticstack_kibana_action_connector" "observability_gemini_connector" {
│ 
╵

```

**Expected behavior**
I would expect the connector to be created. I was able to make it work when I created it from UI

When I created connected in UI, I exported it as saved object and matched the config to Terraform. Thats how I find the `.gemini`  as connector_type_id.
```
{
  "attributes": {
    "actionTypeId": ".gemini",
    "config": {
      "apiUrl": "https://europe-west3-aiplatform.googleapis.com",
      "defaultModel": "gemini-1.5-pro-001",
      "gcpProjectID": "<redacted>",
      "gcpRegion": "europe-west3"
    },
    "isMissingSecrets": true,
    "name": "gemini-playground",
    "secrets": {}
  },
  "coreMigrationVersion": "8.8.0",
  "created_at": "2024-08-23T08:22:53.546Z",
  "id": "dddd16c7-859e-49cf-903f-9305f82c660b",
  "managed": false,
  "references": [],
  "sort": [
    1724401373546,
    4294967344
  ],
  "type": "action",
  "typeMigrationVersion": "10.1.0",
  "updated_at": "2024-08-23T08:22:53.546Z",
  "version": "WzIwOTAwOTUsNDdd"
}

```

**Versions (please complete the following information):**
 - OS: Linux
 - Terraform Version: 1.5.7
 - Provider version: 0.11.6
 - Elasticsearch Version: 8.15.0

### PR DESCRIPTION
**Pull request by tehbooom**
Closes #727

### CODE REVIEW

File: generated/connectors/bundled.yaml
Comment (tobio) [inline generated/connectors/bundled.yaml:1]: Were the additions here copied from the current Kibana spec or were they created manually?

Comment (tehbooom) [reply generated/connectors/bundled.yaml:1]: Manually. If this needs to be copied from the spec could you provide link? Thanks.

Comment (tobio) [reply generated/connectors/bundled.yaml:1]: [This](https://github.com/elastic/kibana/tree/19dd65c24458b6586cb314ce59f0a14340dd5d24/x-pack/plugins/actions/docs/openapi) appears to be the last standalone connectors spec. It'd be worth checking if there's any interesting differences in the Gemini definitions, but I'm happy to stick with the manual edit at the moment. 

Based on the readme it looks like there's a bit of work involved in being able to use a vanilla Kibana spec sadly.

Comment (tehbooom) [reply generated/connectors/bundled.yaml:1]: The main differences were the `defaultModel` config property having a default and not being required as well as some of the descriptions. 

> Based on the readme it looks like there's a bit of work involved in being able to use a vanilla Kibana spec sadly.

I went ahead and updated all of those manually from the spec. Do we have another process that is not manual?

Comment (tobio) [reply generated/connectors/bundled.yaml:1]: > Do we have another process that is not manual?

Ideally Kibana would publish working specs

**Comment by tobio** [comment]
@tehbooom are you able to take a look at the test failure in 8.15?

**Comment by tehbooom** [comment]
> @tehbooom are you able to take a look at the test failure in 8.15?

Yes I should be able to get to this today

**Review by dimuon — APPROVED** [review]

**Pull request merged**